azavea/iow-boundary-tool
# Authentication Fixes

## Overview

This PR does mainly three things:
- Prevents the app from rendering without an authenticated user
- Prevents non-contributors from getting to the welcome page
- Prevents contributors from accessing the app without selecting a utility

Closes #168 
Closes #78 
Closes #149 
Closes #164 

### Demo

https://user-images.githubusercontent.com/8356789/198731408-957e1812-71de-4d34-b3ba-63c45d74085b.mov

### Notes

The private routes are moved into an auth guard which prevents render of children until the user is authenticated.

The utility guard might need to moved into the private routes element and surround only certain elements. For example, if a user visits a submission details page, they'll still have to select a utility. 

## Testing Instructions

- Go to http://localhost:4545
- Log in
- Navigate around
- Refresh a page, a center spinner should be visible breifly before the app loads
- You should not have had to log in again.
- Log out
- Go to a private page
- You should be redirected to the login page

## Checklist
